### PR TITLE
feat: benchmark document construction

### PR DESCRIPTION
**Pull request by JoanFM**

### CODE REVIEW

**Comment by maateen** [comment]
Please don't forget to add python dependencies to `requirements.txt` file.

**Comment by JoanFM** [comment]
> Please don't forget to add python dependencies to `requirements.txt` file.

Which dependencies do u see?

**Review by nan-wang — APPROVED** [review]
LGTM👍

**Comment by maateen** [comment]
> Which dependencies do u see?

Does `numpy` come with `jina` itself? Sorry, I am not sure.

File: src/timecontext.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+class TimeContext:
+    """Timing a code snippet with a context manager."""
+
+    def __enter__(self):
+        self.start = time.perf_counter()
+        return self
+
+    def __exit__(self, typ, value, traceback):
+        self.duration = time.perf_counter() - self.start
```
Comment (bwanglzu) [inline src/timecontext.py:12]: why not use jina logger time context?

Comment (JoanFM) [reply src/timecontext.py:12]: It is noisy and verbose

**Pull request merged**